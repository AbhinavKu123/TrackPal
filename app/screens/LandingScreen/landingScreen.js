import React, { Component } from "react";
import { View, Text, ScrollView, ImageBackground } from "react-native";
import styles from "./style.js";
import { WaveIndicator } from "react-native-indicators";

export default class LandingScreen extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showSpinner: true
    };
  }
  render() {
    return (
      <ImageBackground
        source={require("../../images/landing.jpg")}
        style={{ width: "100%", height: "100%", flex: 1 }}
      >
        <View style={styles.firstContainer}>
          <ScrollView style={styles.scrollStyle}>
            <View style={styles.container2}>
              <Text>Logo Goes Here</Text>
              <WaveIndicator color="black" style={styles.indicator} />
            </View>
          </ScrollView>
        </View>
      </ImageBackground>
    );
  }
}
