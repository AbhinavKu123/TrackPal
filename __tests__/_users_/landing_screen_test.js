import React from 'react';
import LandingScreen from '../../app/screens/LandingScreen/landingScreen.js';

import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer.create(<LandingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});