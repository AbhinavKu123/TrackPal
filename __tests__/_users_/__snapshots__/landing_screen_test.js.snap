// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/landing.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.7)",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.5,
              "justifyContent": "flex-start",
              "marginBottom": 30,
              "marginTop": 266.8,
            }
          }
        >
          <Text>
            Logo Goes Here
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "margin": 10,
                  "marginTop": 50,
                },
              ]
            }
          >
            <View
              animating={true}
              animationDuration={1600}
              animationEasing={[Function]}
              color="black"
              interaction={true}
              renderComponent={[Function]}
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              waveFactor={0.54}
              waveMode="fill"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 40,
                      "opacity": 1,
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                      "width": 40,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 40,
                      "opacity": 1,
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                      "width": 40,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 40,
                      "opacity": 1,
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                      "width": 40,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 40,
                      "opacity": 1,
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                      "width": 40,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
